pkg/store/cache/filter_cache.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package storecache
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/oklog/ulid"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/storage"
+	"golang.org/x/exp/slices"
+)
+
+type FilteredIndexCache struct {
+	cache        IndexCache
+	enabledItems []string
```

Comment:
Maybe this should be a map, that way we won't need to call slices.contains everywhere

Reply:
Since we can only have 3 options, `slices.contains` should be O(1) so might be even faster than using a hashmap here